planner/core/find_best_task.go

```diff
@@ -1118,10 +1118,6 @@ func (ds *DataSource) findBestTask(prop *property.PhysicalProperty, planCounter
 		}
 		var hashPartColName *model.CIStr
 		if tblInfo := ds.table.Meta(); canConvertPointGet && tblInfo.GetPartitionInfo() != nil {
-			// We do not build [batch] point get for dynamic table partitions now. This can be optimized.
-			if ds.ctx.GetSessionVars().StmtCtx.UseDynamicPartitionPrune() {
-				canConvertPointGet = false
-			}
 			if canConvertPointGet && len(path.Ranges) > 1 {
 				// We can only build batch point get for hash partitions on a simple column now. This is
```

Comment:
Please create a follow-up enhancement issue, for adding the support for Range and List partitioned tables as well. Since it currently looks like it only support Key and Hash partitioned tables.

Reply:
Opened https://github.com/pingcap/tidb/issues/45532